### PR TITLE
Initialize LLM validator + basic QA example

### PR DESCRIPTION
**Pull request by dannyadkins**
This change ports over basic llm_validator functionality.

Try it out with:
`bun examples/validator/index.ts`

CleanShot 2024-01-03 at 17 57 25@2x

TODO: 
- [x]  add a test 
- [x]  add small readme so we can carry that over to docs when its ready 
- [x]  merge main

### CODE REVIEW

File: examples/validator/index.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import { LLMValidator } from "@/validator"
+import OpenAI from "openai"
+import { z } from "zod"
+
+const QuestionAnswerSchema = z.object({
+  question: z.string(),
+  answer: z.string()
```
Comment (jxnl) [inline examples/validator/index.ts:7]: ?is there a way to attach the validator to a property of answer

Comment (dannyadkins) [reply examples/validator/index.ts]: hmm u mean like this right? 

```
class QuestionAnswerNoEvil(BaseModel):
    question: str
    answer: Annotated[
        str,
        BeforeValidator(
            llm_validator("don't say objectionable things", allow_override=True)
        ),
    ]
  ```
  
i am p sure yeah -- we can do like this in zod: 
```
const questionAnswerSchema = z.object({
  question: z.string(),
  answer: z.string().refine((answer) => {
 ... 
  }, {
    message: "custom error" 
  }),
});
```

But gotta go rn and can work on that tmrw.

Comment (dannyadkins) [reply examples/validator/index.ts]: jk lol just added

**Comment by jxnl** [comment]
sorry i broke main lol cause i cant read js

**Comment by dannyadkins** [comment]
@jxnl wdyt here? Can merge or close out

File: examples/validator/index.ts (hunk 1 above)
Comment (dannyadkins) [reply examples/validator/index.ts]: You can do it like so:
```
const QuestionAnswerSchemaNoEvil = z.object({
  question: z.string(),
  answer: z.string().refine(async answer => {
    const validator = new LLMValidator({
      statement: "don't say objectionable things",
      allowOverride: false,
      model: "gpt-3.5-turbo",
      temperature: 0,
      openaiClient: new OpenAI({
        apiKey: process.env.OPENAI_API_KEY ?? undefined,
        organization: process.env.OPENAI_ORG_ID ?? undefined
      })
    })
    await validator.validate(answer)
    return true
  })
})
```

**Comment by jxnl** [comment]
I think the challenges right now the valuator can run on the object but I don't know if the Valor can be run when the LLM creates the object itself does that make sense??

**Comment by jxnl** [comment]
can it be used as part of a create call?

**Comment by jxnl** [comment]
sos like 

```js
resp = create(..., response_mode: {schema: Answer})
```

where Answer has a validator

**Comment by jxnl** [comment]
#76  was merged with a

**Pull request closed without merging**